### ISSUE TITLE
[bench request]: typecastlm — frozen Qwen3.5-4B, computed head, TypeSafe wire format (self-hosted package)

### ISSUE THREAD

**Issue by mihail-gribov** [fstandhartinger/jevbench#72]
**Package:** https://github.com/mihail-gribov/typecastlm at [`e95911e`](https://github.com/mihail-gribov/typecastlm/tree/e95911e) · weights https://huggingface.co/mihailgribov/typecastlm-qwen3.5-3.8b at `dcfecfd`

typecastlm answers the `typesafe` wire format with `Qwen/Qwen3.5-4B`, frozen. The last four blocks
are removed and a 39-row classification head is added: `true`, `false`, `unsure` and 36 answer
marks. **The head rows are computed, not trained** — each is the model's own output row for the
token it reads, taken from the tied embedding; the `unsure` row is a class-mean direction from 900
labelled FEVER rows. There is no gradient descent anywhere in the making of the checkpoint, and
blocks 0–27 are the original weights bit for bit. One forward pass per decision, no tokens
generated, one calibration temperature per question type.

**What we measured with JevBench's CLI at `2fa63fa` on the public set (231 items):**

| | RTX 5060 Ti 16 GB |
|---|---|
| correct | 180/231 |
| easy / standard / hard | 47/48 · 68/72 · 65/111 |
| answered and valid | 231/231 |
| ECE | 0.062 |
| latency p50 / p95 | 51 ms / 519 ms |
| mean input tokens | 664 |

**Identity check:** a correct setup reproduces 180/231 with that tier split.

**To run it** — the service speaks your wire format, so the existing `typesafe` adapter reaches it
with no new code:

```
pip install "typecastlm[server]" flash-linear-attention fla-core
typecastlm-serve --model mihailgribov/typecastlm-qwen3.5-3.8b --port 8000
python3 -m jevbench.cli run \
    --tasks datasets/public/easy.jsonl,datasets/public/original.jsonl,datasets/public/hard.jsonl \
    --adapter typesafe --endpoint http://127.0.0.1:8000 --model typecastlm-qwen3.5-3.8b \
    --key-env "" --results out.jsonl
```

First start downloads 7.5 GB. No hosted endpoint exists and none is planned, so the run is
self-hosted on your pods and the row needs no API flag.

**Disclosures**

- Context: 32768 state tokens. A longer state is **not** refused — the middle is dropped and the
  two halves are kept, so a long item still gets an answer rather than an error.
- Calibration: four temperatures (1.3 verdict, 2.85 three-answer, 1.75 choice, 3.9 scale), fitted
  on FEVER dev, ARC-Challenge and SST-5. The public set was never used for any fit, and the
  wording was selected on those same outside sets, never on yours.
- Options are presented as `A. text` in the order `criteria` lists them, criteria text verbatim.
  The mark is the option's position, so a different order is a different reading: reordering the
  options of the 213 option questions here flips 29 verdicts. Your adapter passes `criteria`
  through unchanged, which is what the numbers above were taken with.
- `score` levels are marked with their own digits when those are single characters, otherwise
  `0…9A…Z`.
- Structured state is rendered with `json.dumps(..., ensure_ascii=False)`.
- Price basis: self-hosted bf16 on one 16 GB consumer card, input tokens only. Nothing is
  generated, so `usage.output_tokens` is always 0.
- Speeds are single-request with no other load and `flash-linear-attention` installed; without
  those kernels p50 triples.
- Every answer carries the raw `logits` as an extra field beside the fields your adapter reads.
  Ignoring it changes nothing.

**Row details**

- **repo:** https://github.com/mihail-gribov/typecastlm
- **author:** Mikhail Gribov
- **underlying:** `Qwen/Qwen3.5-4B` (Alibaba Cloud), frozen through block 27, no fine-tuning. The
  head, the wording and the calibration are ours.
- **licence:** Apache-2.0 for the package and the weights; `NOTICE` states what the derivative
  changed, as §4(b) asks.

The sealed items are yours to run — we have never seen them and nothing is fitted to them.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold
